src/Tasks/Microsoft.NET.Build.Tasks/PrepareForReadyToRunCompilation.cs

```diff
@@ -79,40 +93,105 @@ protected override void ExecuteCore()
             var supportedRIDsList = supportedRuntimeIdentifiers == null ? Array.Empty<string>() : supportedRuntimeIdentifiers.Split(';');
 
             // Get the best RID for the host machine, which will be used to validate that we can run crossgen for the target platform and architecture
-            string hostRuntimeIdentifier = NuGetUtils.GetBestMatchingRid(
+            _hostRuntimeIdentifier = NuGetUtils.GetBestMatchingRid(
                 runtimeGraph,
                 NETCoreSdkRuntimeIdentifier,
                 supportedRIDsList,
                 out bool wasInGraph);
 
-            if (hostRuntimeIdentifier == null || _runtimeIdentifier == null || _packagePath == null)
+            if (_hostRuntimeIdentifier == null || _targetRuntimeIdentifier == null)
             {
                 Log.LogError(Strings.ReadyToRunNoValidRuntimePackageError);
                 return;
             }
 
-            if (!ExtractTargetPlatformAndArchitecture(_runtimeIdentifier, out string targetPlatform, out _targetArchitecture) ||
-                !ExtractTargetPlatformAndArchitecture(hostRuntimeIdentifier, out string hostPlatform, out Architecture hostArchitecture) ||
+            if (ReadyToRunUseCrossgen2)
+            {
+                if (!ValidateCrossgen2Support())
+                    return;
```

Comment:
nit: braces

Reply:
ok, I give up on commenting on all of them. :)

Reply:
Added braces to a bunch of places